src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/AddKeepAlivesStep.cs

```diff
@@ -103,24 +103,51 @@ bool AddKeepAlives (TypeDefinition type)
 					if (method.Parameters [i].ParameterType.IsValueType || method.Parameters [i].ParameterType.FullName == "System.String")
 						continue;
 
-					changed = true;
-
 					if (methodKeepAlive == null)
-						methodKeepAlive = Context.GetMethod (
-#if NETCOREAPP
-							"System.Private.CoreLib",
-#else
-							"mscorlib",
-#endif
-							"System.GC", "KeepAlive", new string [] { "System.Object" });
+						methodKeepAlive = GetKeepAliveMethod ();
+
+					if (methodKeepAlive == null) {
+						LogMessage ("Unable to add KeepAlive call, did not find System.GC.KeepAlive method.");
```

Comment:
@dellis1972: *Should* this either be a `LogDebugMessage()` or a `LogWarning()` with new warning code, or is a "normal" `LogMessage()` acceptable?

Reply:
Never mind, wrong `LogMessage()`…

Reply:
If its not actionable by the user, the `LogDebugMessage` might be better. `LogMessage` has a cost since it will emit in any build except quiet. Where the LogDebug* works only for diagnostic builds (and bin logs)